code/game/gamemodes/clock_cult/clock_structures.dm

```diff
@@ -0,0 +1,765 @@
+	toggle() //Toggles off as soon as it's created, but starts online for reasons
+
+/obj/structure/clockwork/mending_motor/Destroy()
+	SSobj.processing -= src
+	..()
+
+/obj/structure/clockwork/mending_motor/examine(mob/user)
+	..()
+	if(is_servant_of_ratvar(user) || isobserver(user))
+		user << "It has [stored_alloy]/[max_alloy] units of replicant alloy."
+
+/obj/structure/clockwork/mending_motor/process()
+	if(!active)
+		return 0
+	if(!stored_alloy && uses_alloy)
+		visible_message("<span class='warning'>[src] emits an airy chuckling sound and falls dark!</span>")
+		toggle()
+		return 0
+	for(var/atom/movable/M in range(5, src))
+		if(istype(M, /mob/living/simple_animal/hostile/anima_fragment))
+			var/mob/living/simple_animal/hostile/anima_fragment/F = M
+			if(F.health == F.maxHealth || F.stat)
+				continue
+			F.adjustBruteLoss(-15)
+			if(uses_alloy)
+				stored_alloy = max(0, stored_alloy - 2)
+		else if(istype(M, /mob/living/simple_animal/hostile/clockwork_marauder))
+			var/mob/living/simple_animal/hostile/clockwork_marauder/E = M
+			if((E.health == E.maxHealth && !E.fatigue) || E.stat)
+				continue
+			E.adjustBruteLoss(-E.maxHealth) //Instant because marauders don't usually take health damage
+			E.fatigue = max(0, E.fatigue - 15)
+			if(uses_alloy)
+				stored_alloy = max(0, stored_alloy - 2)
+		else if(istype(M, /mob/living/silicon))
+			var/mob/living/silicon/S = M
+			if(S.health == S.maxHealth || S.stat == DEAD || !is_servant_of_ratvar(S))
+				continue
+			S.adjustBruteLoss(-25)
+			S.adjustFireLoss(-25)
+			if(uses_alloy)
+				stored_alloy = max(0, stored_alloy - 5) //Much higher cost because silicons are much more useful
+		else if(istype(M, /obj/structure/clockwork))
+			var/obj/structure/clockwork/C = M
+			if(C.health == C.max_health)
+				continue
+			C.health = min(C.health + 10, C.max_health)
+			if(uses_alloy)
+				stored_alloy = max(0, stored_alloy - 1)
+	return 1
+
+/obj/structure/clockwork/mending_motor/attack_hand(mob/living/user)
+	if(user.canUseTopic(src)) //Unnecessary?
+		if(!stored_alloy)
+			user << "<span class='warning'>[src] needs more replicant alloy to function!</span>"
+			return 0
+		toggle(user)
+
+/obj/structure/clockwork/mending_motor/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/clockwork/component/replicant_alloy) && is_servant_of_ratvar(user))
+		if(stored_alloy + 10 > max_alloy)
+			user << "<span class='warning'>[src] is too full to accept any more alloy!</span>"
+			return 0
+		user.whisper("Genafzhgr vagb jngre.")
+		user.visible_message("<span class='notice'>[user] liquifies [I] and pours it onto [src].</span>", \
+		"<span class='notice'>You liquify [src] and pour it onto [src], which transfers the alloy into its reserves.</span>")
+		stored_alloy = min(max(0, stored_alloy + 10), max_alloy)
+		user.drop_item()
+		qdel(I)
+		return 1
+	else
+		return ..()
+
+/obj/structure/clockwork/mending_motor/proc/toggle(mob/living/user)
+	if(!user || !is_servant_of_ratvar(user))
```

Comment:
I use that style because if `user` doesn't exist in the first place it ends immediately but if there is then it goes on to check if they're not a servant.